pgcli/packages/expanded.py

```diff
@@ -0,0 +1,41 @@
+def pad(field, total, char=u" "):
+    return field + (char * (total - len(field)))
+
+def get_separator(num, header_len, data_len):
+    total_len = header_len + data_len + 1
+
+    sep = u"-[ RECORD {0} ]".format(unicode(num))
+    if len(sep) < header_len:
+        sep = pad(sep, header_len - 1, u"-") + u"+"
+
+    if len(sep) < total_len:
+        sep = pad(sep, total_len, u"-")
+
+    return sep + u"\n"
+
+def expanded_table(rows, headers):
+    header_len = max([len(x) for x in headers])
+    max_row_len = 0
+    results = []
+
+    padded_headers = [pad(x, header_len) + u" |" for x in headers]
+    header_len += 2
+
+    for row in rows:
+        row_len = max([len(str(x)) for x in row])
```

Comment:
I'd prefer to do 

```
row_len = len(max(row, key=lambda x: len(x)))
```

This prevents creating a new list for each iteration. It might not be a huge deal for small tables, but if someone does this on a table with thousands of rows, it'll probably result in a lot of GC thrash.